### PR TITLE
Implement SELF_PLAY_TILE

### PR DESCRIPTION
**Pull request by yummyblabla**
Implement message sending and receiving for SELF_PLAY_TILE

### CODE REVIEW

File: src/types/responses.ts [hunk 1]
```diff
@@ -158,3 +158,13 @@ export interface PlayedTileInteractionJSON {
...
   connectionId: string;
 }
+
+/**
+ * Interface for SELF_PLAY_TILE payload received
+ */
+export interface SelfPlayTileJSON {
+  connectionId: string;
+  playedTile: string;
+  quad?: boolean;
```
Comment (jzc1828) [inline src/types/responses.ts:168]: I changed this to `isQuad` thinking it might be easier to recognize it as a **boolean**

Comment (yummyblabla) [reply src/types/responses.ts:168]: Sure !

File: src/pages/Game.tsx [hunk 2]
```diff
@@ -71,6 +73,10 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
       const payload = params as Record<string, unknown>;
       ws?.sendMessage(OutgoingAction.PLAYED_TILE_INTERACTION, { gameId: game.gameId, ...payload });
     },
+    [OutgoingAction.SELF_PLAY_TILE]: (params: unknown) => {
+      const payload = params as Record<string, unknown>;
+      ws?.sendMessage(OutgoingAction.PLAYED_TILE_INTERACTION, { gameId: game.gameId, ...payload });
```
Comment (jzc1828) [inline src/pages/Game.tsx:78]: should it be **OutgoingAction.SELF_PLAY_TILE**?
```ts
ws?.sendMessage(OutgoingAction.SELF_PLAY_TILE, { gameId: game.gameId, ...payload });
```

File: src/pages/Game.tsx [hunk 3]
```diff
@@ -171,6 +177,18 @@ const GamePage = ({ ws, currentUser }: GameProps): JSX.Element => {
     const mjPlayer = mjGameState.getMjPlayer();
 
     mjPlayer.addTileToHand(tile);
+
+    // Send SELF_PLAY_TILE if tile is a Bonus Tile
+    if (isBonusTile(data.tile)) {
+      mjPlayer.getHand().setCannotPlayTile();
+      const wsPayload = {
```
Comment (jzc1828) [inline src/pages/Game.tsx:184]: can force it to be the same as the `SelfPlayTileJSON` interface:
```ts
const wsPayload: SelfPlayTileJSON = {
}
```

File: src/pages/Game.tsx (hunk 2 above)
Comment (yummyblabla) [reply src/pages/Game.tsx:78]: Good catch. Error due to copy pasta

File: src/pages/Game.tsx (hunk 3 above)
Comment (yummyblabla) [reply src/pages/Game.tsx:184]: Payload interfaces are for messages incoming to the client. Message validator implicitly? explicitly? checks the keys without using an interface

**Comment by jzc1828** [comment]
need to rebase master

**Pull request merged**